### ISSUE TITLE
Multiple label matcher refused by admission webhook

### ISSUE THREAD

**Issue by lbesson-della** [prometheus-operator/prometheus-operator#3661]
**What did you do?**
Create a PrometheusRule with a multi label matcher expression, e.g. `http_requests_total{job="prometheus",group="canary"}`

The rule is refused by the admission webhook "prometheusrulemutate.monitoring.coreos.com" denied the request: Rules are not valid

**Did you expect to see some different?**

**Environment**

* Prometheus Operator version:

v0.38.1

* Kubernetes version information:

   $ kubectl version                                                                                                                                                                                                                              
Client Version: version.Info{Major:"1", Minor:"19", GitVersion:"v1.19.1", GitCommit:"206bcadf021e76c27513500ca24182692aabd17e", GitTreeState:"clean", BuildDate:"2020-09-09T11:26:42Z", GoVersion:"go1.15", Compiler:"gc", Platform:"windows/am
d64"}                                                                                                                                                                                                                                          
Server Version: version.Info{Major:"1", Minor:"17", GitVersion:"v1.17.11", GitCommit:"3a3612132641768edd7f7e73d07772225817f630", GitTreeState:"clean", BuildDate:"2020-09-02T06:46:33Z", GoVersion:"go1.13.15", Compiler:"gc", Platform:"linux/
amd64"}                                                                                                                                                                                                                                        
                                                                                                                                                                                                                                               

* Kubernetes cluster kind:

    The cluster is managed via terraform, the prometheus oeprator is installed with helm

* Manifests:

```
apiVersion: monitoring.coreos.com/v1
kind: PrometheusRule
metadata:
  name: myservice.rules
  namespace: default
  labels:
    prometheus: kube-prometheus
spec:
  groups:
  - name: myservice-metric
    rules:
    - record: myservice_metric
      expr: rabbitmq_queue_messages{queue="myservice",job="prometheus-rabbitmq-exporter-0"}
      labels:
        namespace: default
        service: rabbitmq
```

* Prometheus Operator Logs:

I've found nothing relevant in the logs

* Anything else?

If I keep only one label, everything works fine

**Comment by simonpasquier** [comment]
Can you try with a more recent version of prometheus-operator? There should be a log message in prometheus-operator.

**Comment by lbesson-della** [comment]
I tried with a more recent version, and I do find logs

`component=admissionwebhook msg="Invalid rule" err="1:61: group \"myservice-metric\", rule 1, \"myservice_metric\": could not parse expression: 1:50: parse error: unexpected end of input inside braces"`

I'm gonna look, but I'm not sure to understand why my expression is refused.

**Comment by simonpasquier** [comment]
There might be a glitch in the expression string. Maybe try wrapping it within single quotes?

_Salut Lionel ! Sympa de te voir ici, ça change de [c2c](https://www.camptocamp.org/profiles/4049/fr/simonp) :)_

**Comment by lbesson-della** [comment]
Using single quotes didn't help, but I solved it with some escaping.

To be complete, I'm passing the rule via terraform, so I was escaping the double quotes. I witched to single quotes, and I tried to add a backslash before the comma and that made it. 
(so the string passed via terraform is  `"rabbitmq_queue_messages{queue='myservice'\\,job='prometheus-rabbitmq-exporter-0'}"`)

I don't know exactly why the escaping of the comma is needed, at least it does the trick and the expression is correct when I describe the PrometheusRule with kubectl.

I tried to apply the rule directly from a yaml file and there was no issue, so the problem doesn't seem to be related to prometheus-operator anyway.

_je me demandais si c'était le même :)_
